cognite_toolkit/_cdf_tk/validation/warning/_base.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import annotations
+
+import dataclasses
+import itertools
+from abc import ABC
+from collections import UserList
+from collections.abc import Collection
+from dataclasses import dataclass
+from functools import total_ordering
+from typing import Any, Generic, TypeVar
+
+
+@total_ordering
+@dataclass(frozen=True)
+class ToolkitWarning(ABC):
```

Comment:
This is the base class for all warnings. 

I implemented support for
* sorting
* frozen=True - makes it hashable which is useful when you can get duplicated warnings
* grouping which is useful for nice user output.

Reply:
We could add support for dumping as a dict and/or dump to json/yaml, if we need it.